x-pack/plugins/apm/server/lib/service_map/get_service_map.ts

```diff
@@ -85,21 +86,27 @@ async function getServicesData(options: IEnvOptions) {
     searchAggregatedTransactions,
   });
 
-  const { filter } = projection.body.query.bool;
+  let { filter } = projection.body.query.bool;
+
+  if (options.serviceName) {
+    filter = filter.concat({
+      term: {
+        [SERVICE_NAME]: options.serviceName,
+      },
+    });
+  }
+
+  if (options.environment) {
+    filter = filter.concat(getEnvironmentUiFilterES(options.environment));
+  }
```

Comment:
@elastic/apm-ui this fixes the issue described [here](https://github.com/elastic/kibana/issues/77573)